### DISCUSSION TITLE
TDEP/IFC Clarifying Question

### DISCUSSION THREAD

**Discussion by ejmeitz** [ttadano/alamode#132] [Q&A] [answered]
I have a clarifying question about TDEP. The IFC calculation package (alm) implements TDEP as far as I can tell, just under a different name. The first step of the IFC calculation is to optimize your structure to remove any residual forces. This necessarily puts you at the bottom of the energy potential which is the 0K equilibrium position. The TDEP [paper ](https://journals.aps.org/prb/abstract/10.1103/PhysRevB.87.104111)updates the position of the atoms so that the equilibrium is no longer the 0K position? I do not see this implemented in ALAMODE...am I missing something?

**Comment by ttadano** [comment, marked as answer]
The "TDEP" method implemented in ALM does not support the relaxation of internal coordinates at finite temperatures. The method explained in the original TDEP paper may indeed work, but I have not  confirmed it by myself. 

If you want to consider this effect using ALM, the following procedure would be necessary:

1. Compute ensemble average of <x_i> from AIMD snapshots
2. Symmetrize <x_i> so that it recovers the original point group symmetry of x0_i
3. Compute \Delta x_i = <x_i> - x0_i (thermal shift)
4. Subtract \Delta x_i from the displacement data in `DFSET`
5. Fit the harmonic force constants to the `DFSET` data

We do not need to remove the residual forces in `DFSET` because they does not affect the 2nd-order force constant values. If you want to fit the 3rd-order force constants at the same time, the residual forces need to be removed, which, in principle, can be done by adding the linear term to the fitting function.
